lib/orientdb_schema_migrator/migration.rb

```diff
@@ -40,33 +40,31 @@ def self.rename_class old_name, new_name
         ODBClient.command "alter class #{old_name} name #{new_name}"
         return true
       else
-        puts "#{class_name} doesn't exist"
         return false
       end
     end
 
-    def self.add_property class_name, property_name, type,property_options={}
+    def self.add_property class_name, property_name, type, property_options={}
+      assert_class_exists!(class_name)
       ODBClient.create_property class_name,property_name,type, property_options
     end
 
     def self.drop_property class_name, property_name
+      assert_class_exists!(class_name)
+      assert_property_exists!(class_name, property_name)
       ODBClient.command "drop property #{class_name}.#{property_name}"
     end
 
     def self.alter_property class_name, property_name, attribute_name, new_value
+      assert_class_exists!(class_name)
+      assert_property_exists!(class_name, property_name)
       ODBClient.command "alter property #{class_name}.#{property_name} #{attribute_name} #{new_value}"
     end
 
-    def self.add_index class_name, index_name, property_name, type
-      # schema has to exist.
-      # check if property exists first, if not ask to create property.
-      if property_exists? class_name, property_name
-        ODBClient.command "create index #{index_name} on #{class_name} (#{property_name}) #{type}"
-        return true
-      else
-        puts "please create property #{property_name} first"
-        return false
-      end
+    def self.add_index class_name, property_name, index_name, type
+      assert_class_exists!(class_name)
+      assert_property_exists!(class_name, property_name)
+      ODBClient.command "create index #{index_name} on #{class_name} (#{property_name}) #{type}"
     end
 
     def self.drop_index index_name
```

Comment:
Good question(s) -- I think ODB doesn't care in these situations, but I will double check.